tests/utilities/test_distributed.py

```diff
@@ -0,0 +1,67 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from unittest import mock
+
+import pytest
+
+
+@mock.patch.dict(os.environ, {"SLURM_PROCID": "0"})
+def test_rank_zero_slurm():
+    """ Test that SLURM environment variables are properly checked for rank_zero_only. """
+    from pytorch_lightning.utilities.distributed import _get_rank, rank_zero_only
+    rank_zero_only.rank = _get_rank()
+
+    @rank_zero_only
+    def foo():
+        # The return type is optional because on non-zero ranks it will not be called
+        return 1
+
+    x = foo()
+    assert x == 1
+
+
+@mock.patch.dict(os.environ, {"RANK": "0"})
+def test_rank_zero_torchelastic():
+    """ Test that torchelastic environment variables are properly checked for rank_zero_only. """
+    from pytorch_lightning.utilities.distributed import _get_rank, rank_zero_only
+    rank_zero_only.rank = _get_rank()
+
+    @rank_zero_only
+    def foo():
+        # The return type is optional because on non-zero ranks it will not be called
+        return 1
+
+    x = foo()
+    assert x == 1
```

Comment:
why separate these 2 from the bottom one?

Reply:
we could parametrize the dict and then within the test do the patch with a context manager

```python
with mock.patch.dict(os.environ, env_dict):
    ...
```

Reply:
n00b question: how would we distinguish what the return should be with the parameterized dict? wouldn't we be re-implementing what's inside of `_get_rank()` ?

Reply:
it looks like the two tests are identical and only differ in the env variable. 
so at the moment it looks like there is no need to distinguish the output.

```python
@pytest.parametrize("env_vars", [{"RANK": "0"}, {"SLURM_PROCID": "0"}])
def test_rank_zero_known_cluster_envs():
    """ Test that torchelastic environment variables are properly checked for rank_zero_only. """

    with mock.patch.dict(os.environ, env_vars):
        from pytorch_lightning.utilities.distributed import _get_rank, rank_zero_only
        rank_zero_only.rank = _get_rank()

        @rank_zero_only
        def foo():
            # The return type is optional because on non-zero ranks it will not be called
            return 1

        x = foo()
        assert x == 1
```

Reply:
oh whoops, thought you meant merging with the test below for `test_rank_zero_none_set`. yes we can merge these 2 tests - I'll do that now